### PR TITLE
Add 110_sm1_model

### PR DESCRIPTION
**Pull request by Carlos-Alegre**
Pull request for the addition of the branch 110_SM1_model

### CODE REVIEW

File: pages/models/generations/SM1/index.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+| --------------| --- | --------- | ------------------- |------------------- | -------- |
+
+
+## Table of references
+
+<a id="1">[1]</a>  Kundur, Prabha. "Power System Stability and Control" New York, USA, 1994, McGraw-Hill
+
+<a id="2">[2]</a>  Kothari, D. P.; Nagrath, I. J. "Modern Power System Analysis", 4th ed., New Dehli, India, 2011, Tata McGraw-Hill
+
+<a id="3">[3]</a>  PowerWorld Corporation. "ECE 310 Syncrhonous Machine Modeling"
```
Comment (matoubongrain) [inline pages/models/generations/SM1/index.md:170]: you can add: Power 
System Dynamics and Stability by Peter Sauer and M.A. Pai from 1998

File: pages/models/generations/SM1/index.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+<p align="center">
+<img src="{{'/pages/models/generations/SM1/SM_rot_model.svg' | relative_url }}"
+     alt="Operation of a Synchronous Machine schematic"
+     style="float: center; width: 400px;" />
+
+</p>
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$v_t = e_f - j i_a X_s - i_a R_a \approx e_f - j i_a X_s$$
```
Comment (matoubongrain) [inline pages/models/generations/SM1/index.md:96]: did you put all the equations of the models? the set of equations you've wrote isn't sufficient to solve the system.

File: pages/models/generations/SM1/index.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$J\frac{2}{p_f}  \frac{d\omega}{dt} = T_m - T_e - T_{fw}$$
+</div>
+
+where $$J$$ is the inertia of the rotating body, $$T_m$$ is the mechanical torque, $$T_e$$ is the electrical torque and $$T_{fw}$$ is the friction and windage torque.
+
+Regarding the electrical part of the model, the rotor winding is magnetized by the exciter current, creating a magnetic field that will rotate with the rotor's angular velocity. As it spins, the stator windings will be subjected to a variable magnetic flux caused by the rotation of the produced magnetic field which, according to Faraday's Law, will induce a current in the windings.
+
+The rotational speed of the rotor of a syncrhonous machine is proportional to the grid frequency times the number of poles (i.e. number of different three-phase windings). The induced field $$e_f$$ will lead the terminal voltage $$v_t$$ by a load angle denoted as $$\delta$$.
```
Comment (matoubongrain) [inline pages/models/generations/SM1/index.md:76]: can you explain the convention you use here for dq axis and delta definition?
see slide 5 here: https://www.powerworld.com/files/Synchronous-Machines.pdf

**Review by matoubongrain — CHANGES_REQUESTED** [review]

File: pages/models/generations/SM1/index.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+# Synchronous Machine SM1 Model
+
+## Context
+
+The synchronous machine is one of the most studied component of a power system, being its main source of electrical energy. It is the most common type of generators, and it is present in conventional power plants that use a turbine to transfer the energy of steam into mechanical energy. These generators transform this mechanical energy into electric energy. The mathematical model presented, will allow to perform steady-state and transient analysis of round rotor synchronous machines.[[1]](#1)
+
+## Model use, assumptions, validity domain and limitations
+
+The model is a basic representation of the synchronous machine and can be used to study three-phase synchronous machine for its steady-state. DYnamic studies can be visited in the models SM2 and SM3, which are obtained working the basic equations of this model and the study of the magnetic coupling between the windings.
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: rephrased the assumptions and limitations of the model

File: pages/models/generations/SM1/index.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+A sychronous machine is formed by the following parts:
+
+#### Rotor
+
+The rotor is the moving part of the generator. It is at the end of the rotation axis of the turbine, which is the component that transfers the energy of a moving fluid (normally steam) into mechanical energy.
+As the name indicates, the mechanical energy corresponds to the rotational kinetic energy of the body compossed by the blades and axis of the turbine. <!-- Add inertia information>
+It can be round rotor, commonly used in high rotation speed application, or salient pole, used in lower speed applications (i.e. hydro generation) where multiple pair of poles are needed to work at the grid frequency.[[2]](#2) The windings axis will be pointing in a perpendicular direction to the turbine axis, and will rotate with the rotor. This winding will have a current generated by the exciter that will produce a rotating magnetic field ($$\phi_f$$). 
+
+#### Exciter
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md:50]: extended explanation on exciter's role in voltage control

File: pages/models/generations/SM1/index.md [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+The exciter is the source of power that generates the current flowing through the rotor winding. It applies a DC voltage $$E_f$$ that generates a current in the rotor winding, creating an electromagnet. 
+
+#### Stator
+
+It is the static part of the machine, as its name indicates. This structure gives physical support to the whole machine, while it tipically contains the armature winding. It is a three-phase winding distributed in a circular cavity 120º apart one from each other. The uniform rotation of the magnetic field created by the rotor winding will generate currents at each winding 120º apart in the phasorial space. The instantaneous value for each phase is noted as $$i_a$$, $$i_b$$, and $$i_c$$[[2]](#2).
+
+<p align="center">
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md:62]: added captions to all the figures

File: pages/models/generations/SM1/index.md [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+<img src="{{'/pages/models/generations/SM1/SM_rot_stat.svg' | relative_url }}"
+     alt="Operation of a Synchronous Machine schematic"
+     style="float: center; width: 400px;" />
+
+</p>
+
+#### Governor
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md:71]: Expanded the governor's role in frequency control

File: pages/models/generations/SM1/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: typos corrected

[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: Definition of delta angle in line 286

File: pages/models/generations/SM1/index.md [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$i_a = I_m\cos(2\pi f t)$$
+
+$$i_b = I_m\cos(2\pi f t - \frac{2\pi}{3})$$
+
+$$i_c = I_m\cos(2\pi f t + \frac{2\pi}{3})$$
+</div>
+
+For dynamic studies of the syncrhonous machine, it is convinient to make use of the dq0 transformation, which will yield the expressions of the currents in the rotating reference frame of the AC waveforms[[1]](#1):
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: Advantages of dq0 explained

File: pages/models/generations/SM1/index.md [hunk 9]
```diff
@@ -0,0 +1,170 @@
+---
+layout: page
+title: SM1 Model
+tags: []
+date: 10/04/2024 
+last-updated: 10/04/2024
+---
+
+# Synchronous Machine SM1 Model
```
[the remark this answers is not in the document]
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: The model can represent both GENROU and GENSAL synchronous machine, so far without saturation

File: pages/models/generations/SM1/index.md (hunk 1 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: added

File: pages/models/generations/SM1/index.md (hunk 2 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: the full model should be completed now, working on the full per unit transformation description

File: pages/models/generations/SM1/index.md (hunk 3 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md]: Convention used added in Figure 4, line 235

File: pages/models/generations/SM1/index.md [hunk 10]
```diff
@@ -0,0 +1,383 @@
...
+The dynamic equation that governs the system relates the mechanical torque applied by the turbine and the electrical torque applied by the grid[[3]](#3):
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$J\frac{2}{p_f}  \frac{d\omega_m}{dt} = T_m - T_e - T_{fw} \quad (1)$$
+</div>
+
+where $$J$$ is the inertia of the rotating body in $$kgm^2$$, $$T_m$$ is the mechanical torque, $$T_e$$ is the electrical torque and $$T_{fw}$$ is the friction and windage torque, all in $$Nm$$.
+
+#### Round-rotor synchronous machine in *abc* reference frame
```
Comment (matoubongrain) [inline pages/models/generations/SM1/index.md:89]: split this section in subsections

File: pages/models/generations/SM1/GENROU/index.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+tags: ["#110","Synchronous machine", "RMS", "EMT", "Phasorial", "Rotor", "Stator", "Dynawo", "Opensource", "GENROU", "GENSAL"]
+date: 10/04/2024 
+last-updated: 23/04/2024
+---
+
+# Salient-pole Synchronous Machine
+
+## Context
+
+As explained in the general Synchronous Machine model, the particular systems that can be derived by making some additional assumptions yield systems of equations that can be easily solved computationally. The GENROU model in *abc* reference frame is one of the simplest models, considering the uniformity of the air gap between the stator and rotor to be able to simplify the equations.
```
Comment (matoubongrain) [inline pages/models/generations/SM1/GENROU/index.md]: can you explain the difference with GENSAL?

Comment (matoubongrain) [reply pages/models/generations/SM1/GENROU/index.md]: can you also explain why this simplified model is used in most power system transient stability simulator?

File: pages/models/generations/SM1/GENROU/index.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+The additional assumptions with respect to the base model made in this model are:
+
+- The air gap between the stator and rotor is uniform.
+- The three-phase system is balanced.
+
+It cannot be used to simulate machines with a non-uniform air gap such as the salient-pole synchronous machine. In fact, the round-rotor model is a particular case of the salient-pole model.
+
+## Model Description
+
+### Parameters
```
Comment (matoubongrain) [inline pages/models/generations/SM1/GENROU/index.md:28]: i assume you will complete this part, as you did for gensal?

File: pages/models/generations/SM1/GENROU/index.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+with the phasorial expression being[[4]](#4): 
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$v_a =  R_a i_a + j X_s i_a + e_{af}$$
+</div>
+
+where $$v_a$$ is the voltage at the terminals of the stator winding $$a$$, $$e_f$$ is the induced electromotive force by the magnetic field rotation, $$X_s = X_a + X_l = \omega_e L_s$$ is the total reactance of the generator, which is obtained by adding the armature and the leakage inductance as shown before and multiplied by the electrical rotation speed, $$R_a$$ is the series resistance, and $$i_a$$ is the current through the $$a$$ winding.
+
+This phasorial equation corresponds to the differential equation obtained previously.
```
Comment (matoubongrain) [inline pages/models/generations/SM1/GENROU/index.md]: you should also add a open source implementation table and references. Powerworld at least. if there are the same as for the generic article you can say it.

File: pages/models/generations/SM1/GENSAL/index.md [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+
+# Salient-pole Synchronous Machine
+
+## Context
+
+The model presented is derived from the general model for Synchronous Machine
+
+## Model use, assumptions, validity domain and limitations
+
+The model can be used to perform dynamic studies for salient-pole synchronous machines in the *dq0* reference frame. It is also valid for the round-rotor synchronous machine mode, as it is a particular case of the salient-pole model.
```
Comment (matoubongrain) [inline pages/models/generations/SM1/GENSAL/index.md:17]: which kind of dynamic studies? not suitable for EMT simulations. if the system is balanced, why keeping abc reference frame on the network. Most of software keep the single phase equivalent model of this one.

File: pages/models/generations/SM1/GENSAL/index.md [hunk 15]
```diff
@@ -0,0 +1,224 @@
...
+
+The assumptions made in this model are:
+
+* The three-phase system is balanced
```
Comment (matoubongrain) [inline pages/models/generations/SM1/GENSAL/index.md:21]: can you clearly list the difference with GENROU?

File: pages/models/generations/SM1/GENSAL/index.md [hunk 16]
```diff
@@ -0,0 +1,224 @@
...
+$$ T''_{q0} \frac{d\Psi_{2q}}{dt} = -\Psi_{2q} - E'_d - (X'_q - X_{ls})I_q$$
+$$ \frac{d\delta}{dt} = \omega_m - \omega_s$$
+$$ \frac{2H}{\omega_s} \frac{d\omega_m}{dt} = T_m - (\Psi_d I_q - \Psi_q I_d) - T_{fw}$$
+$$ \Psi_d = -X''_d I_d + (\frac{X'_d - X_{ls}}{X'_d - X_{ls}}) E'_q + (\frac{X'_d - X''_d}{X'_d - X_{ls}}) \Psi_{1d}$$
+$$ \Psi_q = -X''_q I_q - (\frac{X'_q - X_{ls}}{X'_q - X_{ls}}) E'_d + (\frac{X'_q - X''_q}{X'_q - X_{ls}}) \Psi_{2q}$$
+$$ \Psi_0 = -X_{ls} I_0$$
+
+</div>
+
+
```
Comment (matoubongrain) [inline pages/models/generations/SM1/GENSAL/index.md:211]: can you add the open source implementations?

File: pages/models/generations/SM1/GENSAL/index.md (hunk 15 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/GENSAL/index.md:21]: GENROU and GENSAL models without saturation are equivalent. When there are saturation terms, then the GENSAL only has saturation for one of the axis while GENROU has for both q and d axis

File: pages/models/generations/SM1/GENSAL/index.md (hunk 14 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/GENSAL/index.md]: Modified the description to state the studies that can be done with this model

File: pages/models/generations/SM1/GENROU/index.md (hunk 11 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/GENROU/index.md]: GENROU model in abc deleted, as it is the same as the GENSAL when there is no saturattion

File: pages/models/generations/SM1/GENROU/index.md (hunk 12 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/GENROU/index.md]: Model deleted

**Review by matoubongrain — APPROVED** [review]

File: pages/models/generations/SM1/index.md [hunk 17]
```diff
@@ -0,0 +1,293 @@
...
+
+- **GENSAL with saturation**: Same as the previous model, but in this case it is treated slightly different due to the saturation occuring only in the salient-poles axis only, while the previous model considers saturation along both *d* and *q* axis.
+
+- **GENTPF/GENTPJ**: It models the saturation in a different way, including the saturation in the network interface equations, something that the GENSAL/GENROU models do not consider, and it reflects better the saturation effects in the whole machine.
+
+Details can be found here:
+
+{% include subfolderNav.html %}
+
+## Open source implementations
```
Comment (GJCRESYM) [inline pages/models/generations/SM1/index.md:272]: Can you check this?
image

**Review by GJCRESYM — CHANGES_REQUESTED** [review]

File: pages/models/generations/SM1/index.md (hunk 17 above)
Comment (Carlos-Alegre) [reply pages/models/generations/SM1/index.md:272]: Added a missing line break

**Review by GJCRESYM — APPROVED** [review]

**Pull request merged**